### PR TITLE
Chapter 4 review complete

### PR DESCRIPTION
**Pull request by christetreault**
I've completed my review of chapter 4.

### CODE REVIEW

File: src/04-poly.lhs [hunk 1]
```diff
@@ -140,7 +140,7 @@ For example, we can define `Vector`s of a given size `N` as:
 {-@ type VectorN a N = {v:Vector a | vlen v == N} @-}
 \end{code}
 
-\noindent and now use this to type `twoLangs` above as:
+\noindent and now use this to make `twoLangs`, as defined above, safe:
```
Comment (ranjitjhala) [inline src/04-poly.lhs]: shouldn't this be `unsafeTwoLangs` now?

File: src/04-poly.lhs [hunk 2]
```diff
@@ -372,10 +369,10 @@ accepts it. </div>
 \vspace{1.0in}
 
 \begin{code}
--- >>> dotProduct (fromList [1,2,3]) (fromList [4,5,6])
--- 32 
+-- For example:
```
Comment (ranjitjhala) [inline src/04-poly.lhs]: why did you change this? the >>> is a "standard" syntax for if you run this in GHCi it evaluates to ...

Comment (christetreault) [inline src/04-poly.lhs]: I made the change to make it more obvious that that's an example of "given this input, this happens"

I'm not super attached to "for example", but I'm not sure where `>>>` comes from. That's certainly not what my GHCi prompt looks like.

File: src/04-poly.lhs (hunk 1 above)
Comment (christetreault) [inline src/04-poly.lhs]: Geez, you're fast!

Seems reasonable. I saw that, but decided against it for reasons unknown. I'll make the change.

Comment (christetreault) [inline src/04-poly.lhs]: ...and done.

File: src/04-poly.lhs (hunk 2 above)
Comment (christetreault) [inline src/04-poly.lhs]: OK, I've reverted this change. "Under protest" I guess, though it's not really a huge deal to me either way :P
